src/types.rs

```diff
@@ -153,6 +153,8 @@ pub struct Token<'a> {
     raw: &'a RawToken,
     i: &'a SourceMap,
     idx: u32,
+
+    is_range: bool,
```

Comment:
This field would fit better on `RawToken`. Then you wouldn't need the `range_tokens` vector on the `SourceMap` and the lookup to see if a token is a range token.

Reply:
Storing `is_range` in `RawToken` makes the memory layout different and this change a breaking change.

Reply:
The memory layout of the struct isn't guaranteed, so changing it would not be a breaking change. However, adding a field to an "open" struct _is_ a breaking change, so you're still right.

Reply:
I forgot submitting my review, sorry.